skynet/modules/stt/vox/app.py

```diff
@@ -0,0 +1,72 @@
+import base64
+
+from fastapi import FastAPI, WebSocket, WebSocketDisconnect
+
+from skynet.logs import get_logger
+from skynet.modules.stt.streaming_whisper.utils import utils
+from skynet.modules.stt.vox.connection_manager import ConnectionManager
+from skynet.modules.stt.vox.decoder import PcmaDecoder
+
+log = get_logger(__name__)
+
+ws_connection_manager = ConnectionManager()
+app = FastAPI()
+decoder = PcmaDecoder()
+
+
+@app.websocket('/ws')
+async def websocket_endpoint(websocket: WebSocket, auth_token: str | None = None):
+    session_id = utils.Uuid7().get()
+    await ws_connection_manager.connect(websocket, session_id, auth_token)
+
+    data_map = dict()
+    sampling_rate = 8000
+
+    while True:
+        try:
+            ws_data = await websocket.receive_json()
+
+            event = ws_data.get('event')
+
+            if event == 'start':
+                sampling_rate = ws_data['start']['mediaFormat']['sampleRate']
+
+            if event == 'media':
+                media = ws_data['media']
+                participant_id: str = media['tag']
+
+                if participant_id not in data_map:
+                    header = (participant_id.encode() + '|en'.encode()).ljust(60, b'\0')
+                    data_map[participant_id] = dict(raw=b'', chunks=0, header=header)
+
+                payload = base64.b64decode(media['payload'])
+                participant = data_map[participant_id]
+
+                participant['raw'] += payload
+                participant['chunks'] += 1
+
+                if participant['chunks'] == 50:  # 50 chunks = 1 second
+                    frames = decoder.decode(participant['raw'], media['timestamp'])
+
+                    decoded_raw = b''
+                    for frame in frames:
+                        decoded_raw += bytes(frame.planes[0])
+
+                    if sampling_rate != 16000:
+                        decoded_raw = utils.resample_audio(decoded_raw, sampling_rate, 16000)
+
+                    await ws_connection_manager.process(
```

Comment:
Doesn't this introduce a backlog, since we are not processing chunks while we are processing them? Once we have a large enough sample it feels like we would trancribe that buffer and process incoming chunks concurrently?